azurelinuxagent/common/osutil/gaia.py

```diff
@@ -37,12 +37,22 @@ class GaiaOSUtil(DefaultOSUtil):
     def __init__(self):
         super(GaiaOSUtil, self).__init__()
 
-    def _run_clish(self, cmd, log_cmd=True):
+    def _run_clish(self, cmd):
+        ret = 0
+        out = ""
         for i in xrange(10):
-            ret, out = shellutil.run_get_output(
-                "/bin/clish -s -c '" + cmd + "'", log_cmd=log_cmd)
-            if not ret:
+            try:
+                final_command = ["/bin/clish", "-s", "-c", "'{0}'".format(cmd)]
```

Comment:
Since this command is opening a new shell, we dont need to tokenise the `cmd` variable.

Eg:
```python
shlex.split("/bin/clish -s -c 'set user admin password-hash *LOCK*'")
['/bin/clish', '-s', '-c', 'set user admin password-hash *LOCK*']
```